### PR TITLE
Fire event on I.say() and add return val to step.passed

### PR DESCRIPTION
**Pull request by hubidu**
- Can we fire events for I.say()
- and add the return value of commands (e. g. from REST requests) to step.passed?

Specifically useful for codepress.

### CODE REVIEW

**Comment by Vorobeyko** [comment]
Can you write tests?

**Comment by DavertMik** [comment]
Seems fine. The only concern I have here is that this is an event for one command only. 
Maybe we can use it like `step.comment`? If so, we can also send steps in gherkin with this event too.

**Comment by hubidu** [comment]
Would be nice to get events for the gherkin steps too. I can add this too.

**Comment by DavertMik** [comment]
@hubidu yeah, let's do that
we can add this event for gherkin features and (maybe?) metasteps. 
It should be called `step.comment` instead of `step.say`

**Comment by hubidu** [comment]
@DavertMik I changed the event name to step.comment and added gherkin support. However I could only implement it in a way that a gherkin step 'step.comment' event is fired AFTER the first codecept step 'step.started' event.
Do you see a way to improve this?

File: lib/actor.js [hunk 1]
```diff
@@ -37,7 +37,10 @@ module.exports = function (obj) {
     });
 
   // add print comment method`
-  obj.say = (msg, color = 'cyan') => recorder.add(`say ${msg}`, () => output.say(msg, `${color}`));
+  obj.say = (msg, color = 'cyan') => recorder.add(`say ${msg}`, () => {
+    event.emit('step.comment', msg);
```
Comment (DavertMik) [inline lib/actor.js]:
```suggestion
    event.emit(event.step.comment, msg);
```
`event.step.comment` should be declared as const inside `events.js`

File: lib/reporter/cli.js [hunk 2]
```diff
@@ -79,6 +79,7 @@ class Cli extends Base {
             printMetaStep(metaStep.metaStep);
           }
           currentMetaStep.unshift(metaStep.toString());
+          event.emit('step.comment', step.metaStep.toString() + (step.metaStep.comment ? step.metaStep.comment : ''));
```
Comment (DavertMik) [inline lib/reporter/cli.js]:
```suggestion
          event.emit(event.step.comment, step.metaStep.toString() + (step.metaStep.comment ? step.metaStep.comment : ''));
```

**Comment by DavertMik** [comment]
> However I could only implement it in a way that a gherkin step 'step.comment' event is fired AFTER the first codecept step 'step.started' event.

Technically, I don't see why they should go in a different order, so it should not be a problem. Maybe what we need is to remove `event.step.started` for gherkin steps at all, replacing it with step.commet. But it will also require some other updates, in allure plugin, for instance

**Comment by DavertMik** [comment]
Ok, I will finialize this

**Pull request merged**